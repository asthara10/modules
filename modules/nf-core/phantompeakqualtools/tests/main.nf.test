nextflow_process {

    name "Test Process PHANTOMPEAKQUALTOOLS"
    script "../main.nf"
    process "PHANTOMPEAKQUALTOOLS"

    tag "modules"
    tag "modules_nfcore"
    tag "phantompeakqualtools"

    test("test_phantompeakqualtools_single_end") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:true ], // meta map
        file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_phantompeakqualtools_paired_end") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
