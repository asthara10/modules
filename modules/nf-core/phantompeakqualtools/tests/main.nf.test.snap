{
    "versions": {
        "content": [
            [
                "versions.yml:md5,e0a48a40af2cf7d5de72c4c3cb47a4fc"
            ]
        ],
        "timestamp": "2023-11-24T14:27:33.14711"
    }
}